zerver/lib/actions.py

```diff
@@ -3799,11 +3799,27 @@ def do_update_message(user_profile: UserProfile, message: Message, topic_name: O
     }  # type: Dict[str, Any]
     changed_messages = [message]
 
+    send_event_users = list()  # call update event for these users
+    ums = UserMessage.objects.filter(message=message.id)
+    ums_user_ids = [user.user_profile_id for user in ums]
     if message.is_stream_message():
         stream_id = message.recipient.type_id
         event['stream_name'] = Stream.objects.get(id=stream_id).name
-
-    ums = UserMessage.objects.filter(message=message.id)
+        # Get subscribers of the stream who are active now.
+        active_subs = Subscription.objects.filter(recipient__type=Recipient.STREAM,
+                                                  recipient__type_id=stream_id,
+                                                  active=True)
+        # Also send update event to subscribers who are not
+        # not in ums, i.e. they subscribed to the stream after
+        # the message was sent, or are not subscribed at all.
+        # TODO: Send update event to active non-subscribers who
+        # can see the message.
+        send_event_users.extend([
+            {
+                'id': sub.user_profile_id,
+                'flags': ['read']
```

Comment:
Reason for settings it as "read"?

Reply:
Because stream's past messages are displayed as read messages.

Reply:
Okay, but I think it should work without this flag as this is kind of mocking. So lets wait for someone else if it is worth to be placed here.